language/go/testutils_test.go

```diff
@@ -0,0 +1,32 @@
+/* Copyright 2017 The Bazel Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package golang
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func equals(t *testing.T, expected, actual interface{}, format string, args ...interface{}) {
+	allTypes := func(_ reflect.Type) bool { return true }
```

Comment:
Can you help me understand why asserting on all types is the correct thing to do here? What fields do you care about that aren't exported? Are they a thing that actually should be a part of the public API?